File: 792-BinarySearch/792-BinarySearch.py
# Last updated: 2/9/2026, 9:54:41 PM
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        left = 0
        right = len(nums) - 1
        
        while left <= right and left >= 0 and right < len(nums):
            middle = (left + right) // 2
            if nums[middle] == target:
                return middle
            elif nums[middle] < target:
                left = middle + 1
            else:
                right = middle - 1
        
        # value doesn't exist
        return -1

File: 792-BinarySearch/792-BinarySearch.java
// Last updated: 2/9/2026, 9:54:43 PM
class Solution {
    int retVal = -1;
    public int search(int[] nums, int target) {
        binarySearch(nums, target);
        return retVal;
    }

    public void binarySearch(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        while(left <= right) {
            int mid = left + (right - left)/2;
            if(target < nums[mid]) {
                right = mid - 1;
            }
            else if (target > nums[mid]) {
                left = mid + 1;
            }
            else {
                retVal = mid;
                break;
            }
        }
    }
}